packages/grid/_modules_/grid/components/cell/GridEditInputCell.tsx

```diff
@@ -33,13 +38,15 @@ export function GridEditInputCell(props: GridCellParams & InputBaseProps) {
       };
 
       if (isDateColumn) {
-        editProps.value = newValue === '' ? null : new Date(newValue); // TODO fix parsing, this is plain wrong.
+        editProps.value = newValue === '' ? null : formatDateString(newValue);
+      } else if (isDateTimeColumn) {
+        editProps.value = newValue === '' ? null : new Date(newValue);
```

Comment:
Should we do the same approach for dateTime? 🤔

Reply:
This is added in the most recent commit